packages/alchemy/src/react/hooks/useSendUserOperation.ts

```diff
@@ -0,0 +1,108 @@
+"use client";
+
+import type {
+  GetEntryPointFromAccount,
+  SendUserOperationParameters,
+  SendUserOperationResult,
+} from "@alchemy/aa-core";
+import {
+  useMutation,
+  type UseMutateAsyncFunction,
+  type UseMutateFunction,
+} from "@tanstack/react-query";
+import type { SupportedAccounts } from "../../config/types.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { ClientUndefinedError } from "../errors.js";
+import type { BaseHookMutationArgs } from "../types.js";
+import { type UseSmartAccountClientResult } from "./useSmartAccountClient.js";
+
+export type UseSendUserOperationMutationArgs<
+  TEntryPointVersion extends GetEntryPointFromAccount<TAccount>,
+  TAccount extends SupportedAccounts = SupportedAccounts
+> = BaseHookMutationArgs<
+  SendUserOperationResult<TEntryPointVersion>,
+  SendUserOperationParameters<TAccount>
+>;
+
+export type UseSendUserOperationArgs<
+  TEntryPointVersion extends GetEntryPointFromAccount<TAccount>,
+  TAccount extends SupportedAccounts = SupportedAccounts
+> = {
+  client: UseSmartAccountClientResult["client"] | undefined;
+  waitForTxn?: boolean;
+} & UseSendUserOperationMutationArgs<TEntryPointVersion, TAccount>;
+
+export type UseSendUserOperationResult<
+  TEntryPointVersion extends GetEntryPointFromAccount<TAccount>,
+  TAccount extends SupportedAccounts = SupportedAccounts
+> = {
+  sendUserOperation: UseMutateFunction<
+    SendUserOperationResult<TEntryPointVersion>,
+    Error,
+    SendUserOperationParameters<TAccount>,
+    unknown
+  >;
+  sendUserOperationAsync: UseMutateAsyncFunction<
+    SendUserOperationResult<TEntryPointVersion>,
+    Error,
+    SendUserOperationParameters<TAccount>,
+    unknown
+  >;
+  sendUserOperationResult:
+    | SendUserOperationResult<TEntryPointVersion>
+    | undefined;
+  isSendingUserOperation: boolean;
+  error: Error | null;
+};
+
+export function useSendUserOperation<
```

Comment:
this is only meant to be used for smart accounts? we should check if account is connected to the client, not just that client exists. in general, throw more informative state error back to the caller.

Makes sense to group signer/smart account/modular account hooks separately in the subdirs of hooks for different accounts, and also have more common state checking/error messages/code shared.

Reply:
The client will throw if it is not connected to an account already and the typing on this specific client parameter forbids it from not not being connected anyways since it has to be a LA client or MA client